include/revng/Pipeline/Runner.h

```diff
@@ -116,6 +125,10 @@ class Runner {
                   const ContainerToTargetsMap &Targets,
                   llvm::raw_ostream *DiagnosticLog = nullptr);
 
+  llvm::Error runAnalyis(llvm::StringRef AnalysisName,
+                         const ContainerToTargetsMap &Targets,
+                         llvm::raw_ostream *DiagnosticLog = nullptr);
```

Comment:
* `runAnalysis`
* `DiagnosticLog` should disappear, either return the information in the `llvm:Error` or in a `Logger`. If you want to do this later, please make sure you do not add any new uses of this.